.github/workflows/e2e.yml

```diff
@@ -0,0 +1,275 @@
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          repository: mattermost/mattermost
+          path: mattermost
+
+      - name: e2e/setup-docker-buildx
+        uses: docker/setup-buildx-action@d70bba72b1f3fd22344832f00baa16ece964efeb # v3.3.0
+
+      - name: e2e/setup-go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version-file: mattermost/server/go.mod
+
+      - name: e2e/generate-configuration
+        working-directory: ./mattermost/server/scripts/config_generator
+        env:
+          OUTPUT_CONFIG: ${{ github.workspace }}/config/config.json
+        run: |
+          mkdir -p ${{ github.workspace }}/config
+          go run main.go
+
+      - name: e2e/patch-generated-configuration
+        env:
+          OUTPUT_CONFIG: ${{ github.workspace }}/config/config.json
+        run: |
+          cp ${OUTPUT_CONFIG} ${OUTPUT_CONFIG}.orig
+          jq -f ${{ github.workspace }}/e2e/overlay-config.jq ${OUTPUT_CONFIG}.orig > ${OUTPUT_CONFIG}
+          diff ${OUTPUT_CONFIG}.orig ${OUTPUT_CONFIG} || true
+
+      - name: e2e/download-mattermost-plugin-calls-package
+        uses: actions/download-artifact@c850b930e6ba138125429b7e5c93fc707a7f8427 # v4.1.4
+        with:
+          name: mattermost-plugin-calls-package
+          path: dist
+
+      - name: e2e/prepare-server
+        env:
+          DOCKER_CLIENT_TIMEOUT: 120
+          COMPOSE_HTTP_TIMEOUT: 120
+          ## Should relative to mattermost/server/build/
+          DOCKER_COMPOSE_FILE: gitlab-dc.postgres.yml
+          DOCKER_COMPOSE_TEST_DATA: ../tests/test-data.ldif
+        run: |
+          mkdir -p ${{ github.workspace }}/logs
+          mkdir -p ${{ github.workspace }}/config
+          mkdir -p ${{ github.workspace }}/dotenv
+
+          ## Navigate to mattermost server build folder to spawn the environment
+          cd mattermost/server/build/
+          ${{ github.workspace }}/e2e/scripts/prepare-server.sh
+          cd -
+
+          ## Add extra environment variables for mattermost server
+          echo "MM_LICENSE=${{ secrets.MM_PLUGIN_CALLS_TEST_LICENSE }}" >> dotenv/app.private.env
+          echo "MM_FEATUREFLAGS_BoardsProduct=true" >> dotenv/app.private.env
+          echo "MM_SERVICEENVIRONMENT=test" >> dotenv/app.private.env
+
+          sudo chown -R 2000:2000 ${{ github.workspace }}/logs
+          sudo chown -R 2000:2000 ${{ github.workspace }}/config
+
+          # Spawn mattermost server
+          echo "Spawning mattermost server ... "
+          docker run -d --quiet --name ${CONTAINER_SERVER} \
+          --net ${DOCKER_NETWORK} \
+          --net-alias mm-server \
+          --user mattermost:mattermost \
+          --env-file="${{ github.workspace }}/dotenv/app.private.env" \
+          -v ${{ github.workspace }}/config:/mattermost/config:rw \
+          -v ${{ github.workspace }}/logs:/mattermost/logs:rw \
```

Comment:
Is there an easy way to access server logs as an artifact? Just asking since it's one of the first thing I used to look at in case of failure.

Reply:
I will create an extra step to upload also server logs in case of failure 👍🏼